tool/doc.dart

```diff
@@ -164,6 +152,24 @@ Future<LintConfig?> fetchConfig(String url) async {
   return processAnalysisOptionsFile(req.body);
 }
 
+Future<Map<String, String>> fetchFixStatusMap() async {
+  if (_fixStatusMap.isNotEmpty) return _fixStatusMap;
+  var url =
+      'https://raw.githubusercontent.com/dart-lang/sdk/main/pkg/analysis_server/lib/src/services/correction/error_fix_status.yaml';
```

Comment:
I have to ask, but would it be better to just load the `FixProcessor` and ask whether each lint's name is a key in the map? Seems like this approach would make it hard to change the structure of the `error_fix_status.yaml` file, and that might cause problems.

Reply:
Great question!

The issue with this approach is that I think we'd have to introduce a dependency on `analysis_server` and we don't want to do this for obvious reasons.  To get around this in the past, I pulled the `FixProcessor` source down and scraped it for this mapping but my sense is that that is at least as brittle as this.

A thought about structural changes: maybe a simple test to sanity-check would be good?  (I'll go ahead and add one to show you what I have in mind.)

Reply:
> I will, however, point out that this is a rather subtle dependency on the package in the sense that it's still depending on a file in that package, it's just that the dependency isn't explicit and it's pinned to HEAD rather than a published version. 

Right.  I agree.

> Of course, I probably wouldn't be bothered about it at all if the linter were in the sdk repo, so maybe that's a signal that I should worry about it now either.

Another good observation.  We have a lot of subtle dependencies already and it's a little uncomfortable to add more.  Having said that, my gut says *this* file is pretty unlikely to change (often) and with a sanity check in place we're sure to notice any breakage before we try and publish new docs that are less complete because this data is no longer available.